sycl/source/detail/scheduler/commands.cpp

```diff
@@ -1773,7 +1773,8 @@ pi_result ExecCGCommand::SetKernelParamsAndLaunch(
   pi_result Error = Plugin.call_nocheck<PiApiKind::piEnqueueKernelLaunch>(
       MQueue->getHandleRef(), Kernel, NDRDesc.Dims, &NDRDesc.GlobalOffset[0],
       &NDRDesc.GlobalSize[0], LocalSize, RawEvents.size(),
-      RawEvents.empty() ? nullptr : &RawEvents[0], &Event);
+      RawEvents.empty() ? nullptr : &RawEvents[0],
+      MQueue->is_event_required() ? &Event : nullptr);
```

Comment:
Could you please clarify why this change is legal? What if a user stores a `sycl::event` object returned by `queue::submit` and call `event::wait` or pass it to the `handler::depends_on`?

Reply:
This is just a test run so that it highlights all the problems and it does not belong to the final solution yet.
So any suggestions for improving this to make it work are welcome!